### ISSUE TITLE
forRoot bug ERROR Function calls are not supported in decorators but 'CacheModule' was called.

### ISSUE THREAD

**Issue by Stradivario** [rxdi/cache#1]
### This is coming from https://github.com/Stradivario/ngx-cache-layer/issues/1
Thanks to @vasilvalkov the issue has being resolved regards my man!
Pull request: https://github.com/Stradivario/ngx-cache-layer/pull/2

versions
``` json
"@angular/core": "^7.2.12",
"@angular/compiler": "^7.2.12",
"@angular/cli": "^7.3.8",
"@angular/compiler-cli": "^7.2.12",
"@rxdi/cache": "^3.0.0"
```

CoreModule:
``` JavaScript
import { CommonModule } from '@angular/common';
import { CacheModule } from '@rxdi/cache';

@NgModule({
  imports: [
    CommonModule,
    CacheModule.forRoot({ localStorage: true })
  ]
})
export class CoreModule {}
```

command
`ng build --prod --aot --extractLicenses=true`

error:

> ERROR in Error during template compile of 'CoreModule'
>   Function calls are not supported in decorators but 'CacheModule' was called.

**Comment by Stradivario** [comment]
@vasilvalkov it is still failing on Angular 7.2.7 we need to find a solution for this

**Comment by vasilvalkov** [comment]
Yes, I have checked it and it fails only with AOT compilation for production and only when using the dist. The src files compile flawlessly in any case.

Is it possible the cause for this error to be in the packaging for distribution as only those files throw error?

**Comment by Stradivario** [comment]
@vasilvalkov I think we are facing this issue 

https://github.com/angular/angular/issues/23609

**Comment by vasilvalkov** [comment]
@Stradivario I tried this approach but no difference. still failing. 
The only thing that worked is when I use the lib by importing the src files. Then, when building with `ng build --prod --aot` goes fine. The moment I import the lib from node_modules (the dist files) and run `ng build --prod --aot` it throws the error. This is the reason I think there is something wrong with the packaging of the source files.
